src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1181,7 +1181,7 @@ private ImmutableArray<BoundExpression> BuildArgumentsForErrorRecovery(AnalyzedA
                             foreach (var parameterList in parameterListList)
                             {
                                 var parameterType = GetCorrespondingParameterType(analyzedArguments, i, parameterList);
-                                if (parameterType?.Kind == SymbolKind.NamedType)
+                                if (parameterType.IsDelegateType() || parameterType.IsExpressionTree())
```

Comment:
It looks like this change also broke some tests because it doesn't handle parameterType == null.